### PR TITLE
Admin Management (Back End)

### PR DESCRIPTION
**Pull request by EricBanker12**
# Description

As an Admin, I would like to upload PDFs, so that users have stories to read

[Trello Card](https://trello.com/c/W71Tol6B/37-admin-upload-pdfs)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Tested Admin Routes
- [x] Tested Middleware

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by EricBanker12** [comment]
[[Admin] - Upload PDFs](https://trello.com/c/W71Tol6B/37-admin-upload-pdfs)

File: src/middleware/jwt/jwt.middleware.ts [hunk 1]
```diff
@@ -2,20 +2,24 @@ import { verify } from 'jsonwebtoken';
...
+        const { parentID, childID, adminID } = verify(
             req.headers.authorization.replace('Bearer ', ''),
             process.env.SECRET_SIGNATURE || 'secret'
         ) as JWT;
 
-        req.user = childID
+        // To Do: fix error Type 'Parent | Child | Admin' not assignable to Type 'Parent | Child'
+        req.user = adminID
+            ? await adminRepo.findOne(adminID)
+            : childID
```
Comment (wSedlacek) [inline src/middleware/jwt/jwt.middleware.ts]: Can you turn this into a switch?

Comment (EricBanker12) [reply src/middleware/jwt/jwt.middleware.ts]: We could do it this way:
```js
switch(adminID ? 1 : childID ? 2 : 3) {
    case 1:
    // code
    break
    case 2:
    // code
    break
    default:
    // code
}
```
But `if` statements might be more readable at that point.
```js
if (adminID) req.user = await adminRepo.findOne(adminID)
else if (childID) req.user = //child
else req.user = //parent
```

Comment (wSedlacek) [reply src/middleware/jwt/jwt.middleware.ts]: What about something more like this?
```js
        switch (adminID || childID || parentID || "fall-through") {
            case adminID:
                req.user = await adminRepo.findOne(adminID);
                break;
            case childID:
                req.user = await childRepo.findOne(childID, { relations: ['parent'] });
                break;
            case parentID:
                req.user = await parentRepo.findOne(parentID, { relations: ['children'] });
                break;
            default:
                req.user = null;
        }
```

Comment (wSedlacek) [reply src/middleware/jwt/jwt.middleware.ts]: https://repl.it/@wSedlacek/SparseMisguidedCollaborativesoftware

**Review by ghost — COMMENTED** [review]
add admin to jwt

**Review by wSedlacek — APPROVED** [review]
Good good prior to the merge

**Pull request merged**